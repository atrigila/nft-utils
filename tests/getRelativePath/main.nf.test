nextflow_pipeline {

    name "Test getRelativePath"
    script "./main.nf"
    tag "getRelativePath"

    test("getRelativePath") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
          // Use getAllFilesFromDir() to get a list of all files and folders from the output directory, minus the timestamped files
          def stable_name = getAllFilesFromDir(params.outdir, true, ['pipeline_info/execution_*.{html,txt}'], null )
          def stable_file = getAllFilesFromDir(params.outdir, false, ['pipeline_info/execution_*.{html,txt}'], null )
          assert snapshot(
            // Use getRelativePath to get the relative path starting from outputDir
            getRelativePath(stable_name, outputDir),
            getRelativePath(stable_file, outputDir)
          ).match()
        }
    }

    test("getAllFilesFromDir & getRelativePath combined") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
          // Use getAllFilesFromDir() with named params to get a list of all files and folders from the output directory, minus the timestamped files
          def stable_name = getAllFilesFromDir(params.outdir, relative: true, ignore: ['pipeline_info/execution_*.{html,txt}'] )
          def stable_file = getAllFilesFromDir(params.outdir, relative: true, includeDir: false, ignore: ['pipeline_info/execution_*.{html,txt}'] )
          assert snapshot(
            stable_name,
            stable_file
          ).match()
        }
    }
}
