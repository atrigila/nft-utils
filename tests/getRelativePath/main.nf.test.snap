{
    "getAllFilesFromDir & getRelativePath combined": {
        "content": [
            [
                "pipeline_info",
                "stable",
                "stable/stable_content.txt",
                "stable/stable_name.txt"
            ],
            [
                "stable/stable_content.txt",
                "stable/stable_name.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.09.0"
        },
        "timestamp": "2024-10-03T20:14:09.332332"
    },
    "getRelativePath": {
        "content": [
            [
                "pipeline_info",
                "stable",
                "stable/stable_content.txt",
                "stable/stable_name.txt"
            ],
            [
                "stable/stable_content.txt",
                "stable/stable_name.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T18:20:38.379963"
    }
}
